tests/TestCase/Controller/ComponentRegistryTest.php

```diff
@@ -209,6 +209,36 @@ public function testUnload()
     }
 
     /**
+     * Test that unloading a none existing component triggers an error.
+     *
+     * This should produce an "Object "Foo" was not loaded before." error
+     * which gets thrown as a \PHPUnit\Framework\Error\Error Exception by PHPUnit.
+     *
+     * @expectedException \PHPUnit\Framework\Error\Error
+     * @expectedExceptionMessage Object "Foo" was not loaded before.
+     * @return void
+     */
+    public function testUnloadUnknown()
+    {
+        $this->Components->unload('Foo');
```

Comment:
I don't know why there isn't any `ObjectRegistry ` so far (or I haven't found it)
Maybe this could be related to testing that the events are being unloaded.
Which I should also test...